modules/apps/data-cleanup/data-cleanup-test/src/testIntegration/java/com/liferay/data/cleanup/internal/upgrade/test/RemoveWidgetLayoutTypeSettingsUpgradeProcessTest.java

```
@@ -0,0 +1,121 @@
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+import com.liferay.portal.kernel.util.UnicodeProperties;
+import com.liferay.portal.kernel.util.UnicodePropertiesBuilder;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.Dictionary;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Georgel Pop
+ */
+@RunWith(Arquillian.class)
+public class RemoveWidgetLayoutTypeSettingsUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testRemoveWidgetLayoutTypeSettings() throws Exception {
+		Layout layout = LayoutTestUtil.addTypeContentLayout(_group);
+
+		UnicodeProperties typeSettingsUnicodeProperties =
+			layout.getTypeSettingsProperties();
+
+		typeSettingsUnicodeProperties.putAll(
+			UnicodePropertiesBuilder.put(
+				LayoutConstants.CUSTOMIZABLE_LAYOUT, "true"
+			).put(
+				LayoutTypePortletConstants.LAYOUT_TEMPLATE_ID, "1_column"
+			).put(
+				"column-1-customizable", "false"
+			).put(
+				"column-2-customizable", "false"
+			).build());
+
+		layout = _layoutLocalService.updateLayout(layout);
+
+		Dictionary<String, Object> properties =
+			HashMapDictionaryBuilder.<String, Object>put(
+				"removeWidgetLayoutTypeSettings", true
+			).build();
+
+		try (ConfigurationTemporarySwapper configurationTemporarySwapper =
+				new ConfigurationTemporarySwapper(
+					_CONFIGURATION_PID, properties)) {
+
+			FinderCacheUtil.clearLocalCache();
+
+			Layout updatedLayout = _layoutLocalService.fetchLayout(
+				layout.getPlid());
+
+			Assert.assertNotNull(updatedLayout);
+
```

Let it fail with an NPE if it is null